src/basicpy/basicpy.py

```diff
@@ -373,3 +378,46 @@ def settings(self) -> Dict:
             current settings
         """
         return self.dict()
+
+    def save_model(self, model_dir: PathLike) -> None:
+        """Save current model to folder.
+
+        Args:
+            model_dir: path to model directory
+
+        Raises:
+            FileExistsError: if model directory already exists"""
+        path = Path(model_dir)
+
+        # NOTE how do we want to handle folder already exists, replace or error?
+        try:
+            path.mkdir()
+        except FileExistsError:
+            raise FileExistsError("Model folder already exists.")
```

Comment:
Let's add an "overwrite" keyword argument and default it to False. Then, in the `except` block throw an error `if not overwrite`.